src/CoWSwapEthFlow.sol

```diff
@@ -122,7 +122,25 @@ contract CoWSwapEthFlow is
     }
 
     /// @inheritdoc ICoWSwapEthFlow
-    function deleteOrder(EthFlowOrder.Data calldata order) external {
+    function deleteOrders(EthFlowOrder.Data[] calldata orderArray) external {
+        for (uint256 i = 0; i < orderArray.length; i++) {
+            // In order to safe gas, there are no events emitted
+            // that would signal whether an order was actually 
+            // successfully deleted. Since solhint-disable-next-line no-unused-vars
+            // is not working with (bool success, ) we declare the variable here.
```

Comment:
This is super clowny. I actually just wanna use:
```solidity
// solhint-disable-next-line all
address(this).call(
                abi.encodeWithSignature(
                    "deleteOrder((address,address,uint256,uint256,bytes32,uint256,uint32,bool,int64))",
                    orderArray[i]
                )
            );
            ```
            But I could not find the right disablers here: 
            https://protofire.github.io/solhint/docs/rules.html#security-rules
            Any suggestion?

Reply:
I believe we shouldn't call the contract itself here as `address(this).call(function)` is more expensive than just calling the function directly.
Unfortunately the only way to do that however is not having the `deleteOrder` function revert in the first place, so we should also implement a different function that doesn't revert. I'd leave this change to another PR.

Reply:
I was thinking about something like [this](https://github.com/cowprotocol/ethflowcontract/compare/non-reverting-cancellation?expand=1) to handle reverts: `deleteOrder` takes a flag, if the flag is set it just stop execution instead of reverting if it would have reverted with `NotAllowedToDeleteOrder`. This is the only case where we want to ignore reverts in `deleteOrders`, since any other revert type is probably telling us that something isn't going as expected (and maybe if the order isn't cancelled but silently fails then it might be retried later, which we don't want).
The main disadvantage is an extra 34 gas of execution cost (unfortunately the optimizer isn't clever enough to remove the trivial if statement). We could also duplicate code, but it's a bit risky for maintenability.

Reply:
Okay, let's do this in another PR